homeassistant/components/tuya/sensor.py

```diff
@@ -35,13 +35,20 @@
     DPType,
     UnitOfMeasurement,
 )
-from .entity import ElectricityTypeData, EnumTypeData, IntegerTypeData, TuyaEntity
+from .entity import (
+    BaseTypeData,
+    ElectricityTypeData,
+    EnumTypeData,
+    IntegerTypeData,
+    TuyaEntity,
+)
 
 
 @dataclass(frozen=True)
 class TuyaSensorEntityDescription(SensorEntityDescription):
     """Describes Tuya sensor entity."""
 
+    raw_type: type[BaseTypeData] = ElectricityTypeData
```

Comment:
I have looked into this, and I think it would be better to default to `None` (not `ElectricityTypeData`)
```suggestion
    raw_type: type[BaseTypeData] | None = None
```
You should then set the `raw_type` to `ElectricityTypeData` for all the existing entity descriptions that have `subkey` defined.

Reply:
I thought about that. Thinking to do it in two steps to prevent breaking existing devices if any is missed. Do you think better do to all now?